src/main/java/edu/cornell/kfs/fp/businessobject/USBankRecordFieldUtils.java

```diff
@@ -0,0 +1,206 @@
+    String theString = line.substring(begin, end);
+    if(theString.trim().length()==0) {
+      return null;
+    }
+    return theString;
+  }
+
+  /**
+   * Extracts a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @param required True if the value is required to not be empty
+   * @param lineCount The current line number
+   * @return The trimmed substring
+   * @throws IOException When a required value is missing
+   * @throws StringIndexOutOfBoundsException When taking the actual substring if the provided bounds are beyond the limits of the superstring
+   */
+  public static String extractNormalizedString(String line, int begin, int end, boolean required, int lineCount) throws IOException {
+    String theValue = extractNormalizedString(line, begin, end);
+    if (required) {
+      if (theValue==null) {
+        throw new IOException("A required value was missing at " + begin + " " + end + " on line " + lineCount);
+      }
+    }
+    return theValue;
+  }
+
+  /**
+   * Extracts a Date from a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @param lineCount The current line number
+   * @return The Date parsed from the trimmed substring
+   * @throws IOException When unable to parse the date
+   */
+  public static Date extractDate(String line, int begin, int end, int lineCount) throws IOException {
+    Date theDate;
+    try {
+      String sub = line.substring(begin, end);
+      String year = sub.substring(0,4);
+      String month = sub.substring(4,6);
+      String day = sub.substring(6,8);
+      theDate = Date.valueOf(year+"-"+month+"-"+day);
+    } catch (StringIndexOutOfBoundsException | IllegalArgumentException e) {
+      // May encounter a StringIndexOutOfBoundsException if the string bounds do not match or 
+      // an IllegalArgumentException if the Date does not parse correctly
+      throw new IOException("Unable to parse date from the value " + line.substring(begin,end) + " on line " + lineCount);
+    }
+    return theDate;
+  }
+
+  /**
+   * Extracts a KualiDecimal from a substring less any ending whitespace for a given beginning and ending position.
+   * 
+   * @param line Superstring
+   * @param begin Beginning index
+   * @param end Ending index
+   * @param lineCount The current line number
+   * @return The KualiDecimal parsed from the trimmed substring
+   * @throws IOException When unable to parse the KualiDecimal
+   */
+  public static KualiDecimal extractDecimal(String line, int begin, int end, int lineCount) throws IOException {
+    KualiDecimal theDecimal;
+    try {
+      String sanitized = line.substring(begin, end);
+      sanitized = StringUtils.remove(sanitized, '-');
```

Comment:
Is this method expected to just return the absolute value of a given number, since it seems to be removing the negative sign? The same goes for the extractDecimalWithCents method.